app/src/main/java/com/kickstarter/ui/adapters/CommentsAdapter.kt

```diff
@@ -4,27 +4,62 @@ import android.view.LayoutInflater
 import android.view.ViewGroup
 import androidx.annotation.LayoutRes
 import com.kickstarter.R
+import com.kickstarter.databinding.CommentInitialLoadErrorLayoutBinding
 import com.kickstarter.databinding.ItemCommentCardBinding
 import com.kickstarter.ui.data.CommentCardData
 import com.kickstarter.ui.viewholders.CommentCardViewHolder
 import com.kickstarter.ui.viewholders.EmptyCommentsViewHolder
+import com.kickstarter.ui.viewholders.InitialCommentLoadErrorViewHolder
 import com.kickstarter.ui.viewholders.KSViewHolder
 
 class CommentsAdapter(private val delegate: Delegate) : KSListAdapter() {
     interface Delegate : EmptyCommentsViewHolder.Delegate, CommentCardViewHolder.Delegate
 
+    companion object {
+        private const val SECTION_INITIAL_LOAD_ERROR = 0
+        private const val SECTION_COMMENT_CARD = 1
+    }
+
+    init {
+        resetList()
+    }
+
     @LayoutRes
     override fun layout(sectionRow: SectionRow): Int {
-        return R.layout.item_comment_card
+        return when (sectionRow.section()) {
+            SECTION_COMMENT_CARD -> R.layout.item_comment_card
+            SECTION_INITIAL_LOAD_ERROR -> R.layout.comment_initial_load_error_layout
+            else -> 0
+        }
     }
 
-    fun takeData(comments: List<CommentCardData>) {
+    private fun resetList() {
         clearSections()
-        addSection(comments)
+        insertSection(SECTION_INITIAL_LOAD_ERROR, emptyList<Boolean>())
+        insertSection(SECTION_COMMENT_CARD, emptyList<CommentCardData>())
+    }
+
+    fun takeData(comments: List<CommentCardData>) {
+        resetList()
+        setSection(SECTION_COMMENT_CARD, comments)
+        submitList(items())
+    }
+
+    fun insertPageError() {
+        resetList()
```

Comment:
same as before

Reply:
Removing ``` resetList ``` causes some weird behavior. The previous error page persists even after data arrives

Reply:
fyi: you need to clean specific sections not all of them. Meaning if you need to show the section "SECTION_INITIAL_LOAD_ERROR" send an empty list to the section "SECTION_COMMENT_CARD" and viceversa. For now it's not a problem but I'm gonna have to change it either way on my open PR(https://github.com/kickstarter/android-oss/pull/1284) as I need to present there 2 sections at the same time